### ISSUE TITLE
Unexpected cursor behavior

### ISSUE THREAD

**Issue by tpoche** [RedMadRobot/input-mask-android#65]
We are using your library to provide masks for various types of date and time input fields.  For example, we have a time input field that uses the following mask `"[00]{:}[00]"` for 24 hour time format.  We have noticed that occasionally the cursor returns to the starting position after typing the first two digits of text when it should not be moving at all in this scenario.  Any ideas why this might be happening and how to fix it?

This is how we are configuring the input mask library to work with our EditText fields:
```kotlin
timeText = findViewById(R.id.formItemTextBox)   // get EditText from layout
val mask = "[00]{:}[00]"
val listener = MaskedTextChangedListener(
        mask,
        false,
        timeText,
        null,
        null)
timeText.addTextChangedListener(listener)
```

**Comment by taflanidi** [comment]
Hey @tpoche!

Please refer to the [InputMask vs. autocorrection & prediction](https://github.com/RedMadRobot/input-mask-android#inputmask-vs-autocorrection--prediction) section of our README. Symptoms look the same, so I'd suggest you to try assigning the `textNoSuggestions` or even `textNoSuggestions|textVisiblePassword` as your field's `inputType` and see if it helps.

Please let me know the results.

**Comment by tpoche** [comment]
This appears to have been fixed by upgrading the library version from v2.3.0 to v3.4.4

**Comment by tpoche** [comment]
Btw, I also had similar issues when supporting 12 hour time formats (ie. `[00]{:}[00]{ }[AA]`) that I was able to resolve using @taflanidi 's suggestion of setting the EditText's inputType to `textVisiblePassword`.  Thanks for your help on this!
